vars/runInspector.groovy

```diff
@@ -28,109 +29,228 @@ import java.util.*
+    def urlRegex = /http.*[^}]/
+    def resutlUrl = (getResults =~ urlRegex)
+    resutlUrl = resutlUrl[0]
+    def fullResult = resutlUrl.toURL().text
+    writeFile(file: 'Inspector_test_reults.html', text: fullResult)
+    archiveArtifacts(artifacts: 'Inspector_test_reults.html', allowEmptyArchive: true)
+    def findings = formatedResults(assessmentArn)
+    cleanUp(stackName, body.region, bucketName, fileName)
+    return findings
+}
 
-      // Fail the pipeline if insepctor tests did not pass and flag either set to true or not set
-      def failonfinding = body.get('failonfinding', true)
 
-      if (testPassed >= 1){
-          if (!failonfinding) {
-              println("One or more interpector test(s) failed on the AMI however \'failonfinding\' is set to \'False\' and hence the pipeline has not failed")
-              return testPassed
-          }
-          else {
-              throw new GroovyRuntimeException("One or more interpector test(s) failed on the AMI")
-          }
-      }
-      else {
-          return testPassed
-      }
+
+def checkFail(failon, findings){
+    // Make a list of severity levels with >= 1 nFindings
+    def severties = findings.keySet()
+    def severityFindings = []
+    for (s in severties){
+        if (findings[s].toInteger() > 0 ) {
+            severityFindings.add(s)
+        }
+    }
+
+    def validSeverity = ['']
+    switch(failon) {
+        case 'Informational':
+            validSeverity = ['Informational', 'Low', 'Medium', 'High']
+            break
+        case 'Low':
+            validSeverity = ['Low', 'Medium', 'High']
+            break
+        case 'Medium':
+            validSeverity = ['Medium', 'High']
+            break
+        case 'High':
+            validSeverity = ['High']
+            break
+        case 'Never':
+            validSeveritgroovyy = ['']
+            break
+    }
+    def testPassed = true
+    validSeverity.each { severity ->
+        if (severityFindings.contains(severity)){
+            testPassed = false
+        } else {
+        }
+    }
+    return testPassed
 }
 
 
+def getAgentStatus(String arn) {
+     def client = AmazonInspectorClientBuilder.standard().build()
+     def request = new PreviewAgentsRequest().withPreviewAgentsArn(arn)
+     def response = client.previewAgents(request)
+     def state = response.getAgentPreviews()[0].getAgentHealth()
+     return state
+}
+
+
+def getIstanceStatus(String id) {
+    def client = AmazonEC2ClientBuilder.standard().build()
+    def request = new DescribeInstancesRequest().withInstanceIds(id)
+    def response = client.describeInstances(request)
+    def state = response.getReservations()[0].getInstances()[0].getState().getName()
+    return state
+}
+
+def cleanUp(String stackName, String region, String bucketName, String fileName){
+    // Pull down cloudformaiton stack and bucket hosting cloudformation template
+    println('Cleaning up all created resources')
+    try {
```

Comment:
no need for this try/finally if you're not catching exceptions here

Reply:
This was intended to not fail the pipeline if there was an unforeseen issue with taking the stack down, such that the pipeline would still complete (as this is not vital to the results) and then the user could manually delete the stack/resolve the issue later. Should I still remove this?

Reply:
that's fine, leave that in then